tfhe/src/shortint/engine/server_side/shift.rs

```diff
@@ -53,6 +53,9 @@ impl ShortintEngine {
         ct: &mut Ciphertext,
         shift: u8,
     ) -> EngineResult<Ciphertext> {
+        if !server_key.is_scalar_left_shift_possible(ct, shift) {
+            self.message_extract_assign(server_key, ct)?;
+        }
```

Comment:
below it can be a unchecked call ?

Reply:
I believe even with a clean ct, the shift might not pass

Reply:
ok then